File: FlowCrypt/src/main/java/com/yourorg/sample/api/retrofit/request/NodeRequestWrapper.java
package com.yourorg.sample.api.retrofit.request;

import com.flowcrypt.email.api.retrofit.request.node.BaseNodeRequest;

/**
 * @author DenBond7
 */
public class NodeRequestWrapper<T extends BaseNodeRequest> {
  private int requestCode;
  private T baseNodeRequest;

  public NodeRequestWrapper(int requestCode, T baseNodeRequest) {
    this.requestCode = requestCode;
    this.baseNodeRequest = baseNodeRequest;
  }

  public int getRequestCode() {
    return requestCode;
  }

  public T getRequest() {
    return baseNodeRequest;
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/request/node/NodeRequestWrapper.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.request.node

/**
 * @author DenBond7
 */
//todo-denbond7 It's old code. Need to remove this
class NodeRequestWrapper<T : NodeRequest>(val requestCode: Int, val request: T)
